### DISCUSSION TITLE
Unable to decrypt backup file

### DISCUSSION THREAD

**Discussion by gautamviradiya** [JFC-Group/JF-Customisation#54] [Q&A] [answered]
@itsyourap 
I have spent 2 day to decrypt RSVOTGIEM******_JCOW414.enc  but not get success.
I have get key.txt from router using [https://github.com/JFC-Group/JF-Customisation/blob/main/Instructions/Get-Any-File-From-JF-ONT-Home-Gateway.md](url)

I have tried using [https://github.com/JFC-Group/JF-Customisation/blob/main/Instructions/Alternate-Way-To-Get-Root-Access-JF-ONT-Home-Gateway.md](url)

My openssl version:
OpenSSL 3.2.0 23 Nov 2023 (Library: OpenSSL 3.2.0 23 Nov 2023)

Firmware Version:
SRCMTF1_JCOW414_R2.52.1

Error in CMD in Mac:
openssl aes-128-cbc -d -kfile key.txt  -in RSVOTGIEM053275_JCOW414.enc -out RSVOTGIEM053275_JCOW414.txt
*** WARNING : deprecated key derivation used.
Using -iter or -pbkdf2 would be better.
bad decrypt
00CC59EC01000000:error:1C800064:Provider routines:ossl_cipher_unpadblock:bad decrypt:providers/implementations/ciphers/ciphercommon_block.c:107:

Please give me command what is used and which openssh version.
Please help me to decrypt and encrypt file.

**Comment by itsyourap** [comment]
Can you give me your default router SSID (written on the back of Router box or the default SSID after resetting your router)?

**Reply by gautamviradiya** [reply]
Thanks for keyguesser.py this is decrypted successfully. But key that work from keyguesser is different from key.txt from router.

I have deleted "rm /flash/telnetDisable" file even though telnet is disabled after router restart.

can I install [pihole](https://pi-hole.net/) for ad block?

**Reply by itsyourap** [reply, marked as answer]
Yes! `key.txt` was used quite a long time ago for encryption/decryption. Currently `keyguesser.py` only works.

~You can use pihole but you have to compile it yourself for your router.~

Correction: You can use pihole as long it doesn't modify the root filesystem but you have to compile it yourself for your router.

**Comment by gautamviradiya** [comment]
Right Now I have tried to edit pfrm2.0/user.config file as root but not able to edit because read only file system. How can I edit?

**Reply by itsyourap** [reply]
Well, simple speaking, you cannot.
You need to dump the mtd image then modify the contents and reflash the mtd in your router.
